src/coreclr/jit/llvm.cpp

```diff
@@ -1228,17 +1463,17 @@ int getLocalOffsetAtIndex(GenTreeLclVar* lclVar)
 
         for (unsigned lclNum = 0; lclNum < lclVar->GetLclNum(); lclNum++)
         {
-            LclVarDsc* varDsc = _compiler->lvaGetDesc(lclNum);
+            varDsc = _compiler->lvaGetDesc(lclNum);
             if (!varDsc->lvIsParam)
             {
                 CorInfoType corInfoType = toCorInfoType(varDsc->TypeGet());
-                if (!canStoreTypeOnLlvmStack(corInfoType))
+                if (!canStoreLocalOnLlvmStack(varDsc))
                 {
                     offset = padNextOffset(corInfoType, offset);
                 }
             }
         }
-        offset = padOffset(localCorInfoType, offset);
+        offset = padOffset(toCorInfoType(lclVar->TypeGet()), offset);
```

Comment:
I think I've said it somewhere, but will repeat myself that I think the "righter" way to have this is to:
1) Transform all locals that will live on the shadow stack into `IND/OBJ(SHADOW_STACK_ADDRESS_LCL_VAR + OFFSET)` (in a pre-SSA walk).
2) Then perform a layout pass that sets `lvStkOffs`, which the codegen will then ultimately use.

(Not in this change of course)